pkg/kubectl/metricsutil/metrics_printer.go

```diff
@@ -53,14 +53,36 @@ func NewTopCmdPrinter(out io.Writer) *TopCmdPrinter {
 	return &TopCmdPrinter{out: out}
 }
 
-func (printer *TopCmdPrinter) PrintNodeMetrics(metrics []metricsapi.NodeMetrics, availableResources map[string]v1.ResourceList, noHeaders bool) error {
+func (printer *TopCmdPrinter) PrintNodeMetrics(metrics []metricsapi.NodeMetrics, availableResources map[string]v1.ResourceList, noHeaders bool, sortBy string) error {
 	if len(metrics) == 0 {
 		return nil
 	}
 	w := printers.GetNewTabWriter(printer.out)
 	defer w.Flush()
 
 	sort.Slice(metrics, func(i, j int) bool {
+		if len(sortBy) > 0 {
+			var ui, uj v1.ResourceList
+			err := scheme.Scheme.Convert(&metrics[i].Usage, &ui, nil)
```

Comment:
This will be expensive, since you'll always convert the field upon every invocation. It'll be reasonable to convert those once and sort the converted list.